internal/addrs/parse_target.go

```diff
@@ -181,6 +181,15 @@ func parseResourceInstanceUnderModule(moduleAddr ModuleInstance, allowPartial bo
 		// Starting a resource address with "resource" is optional, so we'll
 		// just ignore it.
 		remain = remain[1:]
+	case "count", "each", "local", "module", "path", "self", "terraform", "var", "template", "lazy", "arg":
```

Comment:
You can target and move modules, which always require the `module` keyword.

Reply:
This function is specifically for parsing resources under modules (as in the modules have already been parsed out). I think it's technically impossible to get here with `module` at the moment as nothing calls this function directly - but it would be an error for a resource address to start with `module`.

I can still take this out, but I think being safe in case anyone references this function directly in future is probably better?

Reply:
oh, then ignore me! 😆 I saw parseTarget under the collapsed code above and missed what this was actually checking.